ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/components/ReportJobs.tsx

```diff
@@ -21,60 +22,75 @@ import { ensureBoolean, ensureStringArray } from 'utils/ensure';
 import ReportRunStatesFilter, { ensureReportRunStates } from './ReportRunStatesFilter';
 import MyJobsFilter from './MyJobsFilter';
 
-function createMockData(scanConfig: ComplianceScanConfigurationStatus) {
-    const snapshots: ComplianceReportSnapshot[] = [
-        {
-            reportJobId: 'ab1c03ae-9707-43d1-932d-f948afb67b53',
-            scanConfigId: scanConfig.id,
-            name: scanConfig.scanName,
-            description: scanConfig.scanConfig.description,
-            reportStatus: {
-                completedAt: '2024-08-27T00:01:40.569402380Z',
-                errorMsg:
-                    "Error sending email notifications:  error: Error sending email for notifier 'fc99e179-57c1-4ba2-8e59-45dbf184c78c': Connection failed",
-                reportNotificationMethod: 'EMAIL',
-                reportRequestType: 'SCHEDULED',
-                runState: 'FAILURE',
-            },
-            reportData: scanConfig,
-            user: {
-                id: 'sso:3e30efee-45f0-49d3-aec1-2861fcb3faf6:c02da449-f1c9-4302-afc7-3cbf450f2e0c',
-                name: 'Test User',
-            },
-            isDownloadAvailable: false,
-        },
-    ];
-    return snapshots;
-}
-
 function getJobId(snapshot: ComplianceReportSnapshot) {
-    return snapshot.scanConfigId;
+    return snapshot.reportJobId;
 }
 
 function getConfigName(snapshot: ComplianceReportSnapshot) {
     return snapshot.name;
 }
 
+const sortOptions = {
+    sortFields: ['Compliance Report Completed Time'],
+    defaultSortOption: { field: 'Compliance Report Completed Time', direction: 'desc' } as const,
+};
+
 type ReportJobsProps = {
-    scanConfig: ComplianceScanConfigurationStatus | undefined;
+    scanConfigId: string;
     isComplianceReportingEnabled: boolean;
 };
 
-function ReportJobs({ scanConfig, isComplianceReportingEnabled }: ReportJobsProps) {
+function ReportJobs({ scanConfigId, isComplianceReportingEnabled }: ReportJobsProps) {
     const { analyticsTrack } = useAnalytics();
 
     const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const { sortOption, getSortParams } = useURLSort(sortOptions);
     const { searchFilter, setSearchFilter } = useURLSearch();
     const [isViewingOnlyMyJobs, setIsViewingOnlyMyJobs] = useURLStringUnion('viewOnlyMyJobs', [
         'false',
         'true',
     ]);
 
-    const filteredReportRunStates = ensureStringArray(searchFilter['Report State']);
+    const filteredReportRunStates = ensureStringArray(searchFilter['Compliance Report State']);
+
+    const query = getRequestQueryStringForSearchFilter({
+        'Compliance Report State': filteredReportRunStates,
+    });
+
+    const fetchComplianceReportHistoryCallback = useCallback(
+        () =>
+            fetchComplianceReportHistory({
+                id: scanConfigId,
+                query,
+                page,
+                perPage,
+                sortOption,
+                showMyHistory: Boolean(isViewingOnlyMyJobs),
```

Comment:
Since `isViewingOnlyMyJobs` is a string, both `Boolean("true")` and `Boolean("false")` will result in `true`.

I think:
```suggestion
                showMyHistory: isViewingOnlyMyJobs === 'true',
```

Reply:
Oops, nice catch. Will fix this 👍🏼